cherab/core/atomic/rates.pyx

```diff
@@ -144,8 +144,13 @@ cdef class BeamCXPEC:
     transition :math:`n\rightarrow n'` of ion :math:`Z^{(\alpha+1)+}` with electron donor
     :math:`H^0` in metastable state :math:`m_{i}`. Equivalent to
     :math:`q^{eff}_{n\rightarrow n'}` in `adf12 <http://open.adas.ac.uk/adf12>_`.
+
+    :param donor_metastable: The metastable state of the donor species for which the rate data applies.
     """
 
+    def __init__(self, int donor_metastable):
+        self.donor_metastable = donor_metastable
```

Comment:
> The classes in core.atomic should be as general as possible. Donor_metastable is a specific which doesn't have to be used by other data sources than ADAS, has it?

The issue with the `donor_metastable` attribute is that it is used in the `BeamCXLine` model:
https://github.com/cherab/core/blob/b3cc6ad8025bbd1a77ab59d64959441572d5f84d/cherab/core/model/beam/charge_exchange.pyx#L338-L341
Thus, the attribute should be in the base class, otherwise any `BeamCXPEC` implementation that does not have this attribute will break the model.

Also, CX PEC is highly dependent on the excited/metastable level of the beam atom, so any atomic data source, not just ADAS, should provide CX PECs resolved over excited/metastable state.

> Also, this change is not related to the original issue with CX PEC.

I wanted to fix two separate issues #441 (wrong ion density) and #411 (missing donor_metastable) with a single PR, since both of these issues affect the same `BeamCXPEC` class.